.github/workflows/publish.yaml

```diff
@@ -1,16 +1,16 @@
 name: Publish Python Package
 
 on:
-  release:
-    types: [published]
+  push:
```

Comment:
This will force us to explicitly tag versions, but we can follow-up with an auto-tagging strategy if needed.